internal/provider/org/role_datasource.go

```diff
@@ -0,0 +1,170 @@
+	Name        types.String `tfsdk:"name"`
+	Description types.String `tfsdk:"description"`
+	BundleKey   types.String `tfsdk:"bundle_key"`
+	ReadOnly    types.Bool   `tfsdk:"read_only"`
+	Rights      types.Set    `tfsdk:"rights"`
+}
+
+func (d *orgRoleDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + "org_role"
+}
+
+func (d *orgRoleDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "The org_role datasource allows you to read roles in an organization.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"name": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "A name for the role",
+			},
+			"description": schema.StringAttribute{
+				// Not ForceNew, to allow the resource name to be updated
+				Computed:            true,
+				MarkdownDescription: "A description for the role",
+			},
+			"bundle_key": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Key used for internationalization",
+			},
+			"read_only": schema.BoolAttribute{
+				Computed:            true,
+				MarkdownDescription: "Indicates if the role is read only",
+			},
+			"rights": schema.SetAttribute{
+				// Not ForceNew, to allow the resource name to be updated
+				Computed:            true,
+				MarkdownDescription: "A list of rights for the role",
+				ElementType:         types.StringType,
+			},
+		},
+	}
+}
+
+func (d *orgRoleDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.CloudAvenue)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *client.CloudAvenue, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+}
+
+func (d *orgRoleDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var (
+		data *orgRoleDataSourceModel
+		err  error
+		role *govcd.Role
+	)
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// role read is accessible only in administrator
+	adminOrg, err := d.client.Vmware.GetAdminOrgByNameOrId(d.client.GetOrg())
+	if err != nil {
+		resp.Diagnostics.AddError("[role read] Error retrieving Org", err.Error())
+		return
+	}
+
+	// Get the role
+	if data.ID.IsNull() {
+		role, err = adminOrg.GetRoleByName(data.Name.ValueString())
+	} else {
+		role, err = adminOrg.GetRoleById(data.ID.ValueString())
+	}
```

Comment:
`ID` is a computed value. 

Add `role_id` and replace `name` to  `role_name` for better understanding.